rundeckapp/grails-app/utils/rundeck/codecs/JobsXMLCodec.groovy

```diff
@@ -600,27 +659,48 @@ class JobsXMLCodec {
                 }
             }
-            map.notification.keySet().sort().findAll { it.startsWith('on') }.each { trigger ->
-                if(map.notification[trigger]){
-                    if(map.notification[trigger]?.email){
-                        def mail= map.notification[trigger].remove('email')
-                        map.notification[trigger].email=[:]
-                        mail.each{k,v->
-                            BuilderUtil.addAttribute(map.notification[trigger].email,k,v)
+            boolean useOldWebhookFormat = useOldFormat(map.notification, true)
+            map.notification.keySet().sort()?.each { trigger ->
+                ArrayList triggerNotifs = map.notification[trigger]
+                map.notification[trigger] = [:]
+                triggerNotifs?.each {notif ->
+                    def xmlNotif = [:]
+                    def xmlNotifType
+                    if(!notif.plugin){
+                        if(notif.email){
+                            notif = notif.email
+                            xmlNotifType = 'email'
+                        }else if(notif.urls){
+                            xmlNotifType = 'webhook'
+                        }else{
+                            throw new JobXMLException("unknown notification xmlNotifType")
                         }
-                    }
-                    if(map.notification[trigger]?.urls){
-                        map.notification[trigger].webhook=BuilderUtil.toAttrMap('urls',map.notification[trigger].remove('urls'))
-                    }
-                    if(map.notification[trigger]?.plugin){
-                        if(map.notification[trigger]?.plugin instanceof Map){
-                            convertNotificationPlugin(map.notification[trigger]?.plugin)
-                        }else if(map.notification[trigger]?.plugin instanceof Collection){
-                            //list of plugins,
-                            map.notification[trigger].plugin.each{Map plugin->
-                                convertNotificationPlugin(plugin)
+                        if(!(useOldWebhookFormat && xmlNotifType=='webhook')){
+                            //for notifications translated to inline tags
+                            notif.each{k,v->
+                                BuilderUtil.addAttribute(xmlNotif,k,v)
                             }
                         }
+                    }else{
+                        notif.plugin.each{Map plugin->
+                            convertNotificationPlugin(plugin)
+                        }
+                        xmlNotifType = 'plugin'
+                        xmlNotif = notif.plugin
+                    }
+                    if(useOldWebhookFormat && xmlNotifType=='webhook'){
+                        map.notification[trigger][xmlNotifType] = BuilderUtil.toAttrMap('urls',notif.remove('urls'))
+                        notif.each { entry ->
+                            xmlNotif = [:]
```

Comment:
i think this line is not needed

Reply:
If it is not present, the webhook is mapped to only 1 xml tag containing 3 attributes (the "new format") instead of 3 different tags.